pkgs/native_assets_builder/CONTRIBUTING.md

```diff
@@ -0,0 +1,43 @@
+`package:native_assets_builder` and `package:native_assets_cli/native_assets_cli_internal.dart` are used in [`package:dartdev`] and [`package:flutter_tools`].
+
+Dartdev and Flutter Tools are the commandline interfaces for `dart ...` and `flutter ...` commands.
+
+## Downstream builds
+
+Both Dartdev and Flutter Tools are built in open source, and inside g3.
+
+* Dartdev as part of the Dart SDK build on the [Dart CI].
+* Dartdev as part of the g3 build (see CBuild comments on CLs on [Gerrit]).
+* Flutter Tools as part of the workflows on GitHub.
+* Flutter Tools as part of the g3 build (see the "Google testing" workflow on PRs on GitHub).
+
+The versions used in these different places are as follows:
+
+* The Dart SDK uses pinned dependencies in [DEPS], the current has is in `native_rev`.
+* The Flutter build on GitHub ussed published dependencies in [its `pubspec.yaml`].
+* The g3 build uses the pinned dependencies that are rolled in from the Dart SDK. **Both for Dartdev _and_ Flutter Tools**.
+
+## Rolling
+
+The above means the following.
+
+1. The DEPS in the Dart SDK can only be rolled forward as much as is still compatible with the published deps currently used in Flutter Tools.
+2. Flutter Tools can only be rolled forward as much as is still compatible with with the deps pinned in the Dart SDK.
+
+So, any breaking change must be done in the following way:
+
+1. Introduce a new API.
+2. Release a published version.
```

Comment:
```suggestion
2. Release a new major version.
```

Reply:
I think we only need to bump the major version when removing the old API right?